lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -251,6 +251,37 @@ sub log_entry {
     );
 }
 
+
+=head2 collapse_future_stack
+
+The future frames are too much and too tedious. This method will keep only one frame if there are many continuously future frames.
+Parameter: log data
+Return: log data
+
+=cut
+
+sub collapse_future_stack{
+    my ($self, $data) = @_;
+    my $stack = $data->{stack};
+    my @new_stack;
+    my $previous_future;
+    for my  $frame ($stack->@*){
```

Comment:
do this `for` loop guarantee that the latest `Future` stack trace will be available after collapsing?

Reply:
It will keep the oldest one (outest), not the latest one (deepest). I'm not sure which one will be better. What do you think ?

Reply:
Well, we should keep the one where the point of failure occurs. I can give an parallel example of null pointer exceptions (uninitialised value error).

```
my $var = undef;
function1($var);

#inside function1:
...
function2($var);
...

#inside function2:
...
function3($var);
...

#inside function3:
...
$var->{key} // point of failure as error occurs here
...
```

The stack trace could be something like
```
at line 29 in module3.pm in function3
at line 57 in module2.pm in function2
at line 17 in module1.pm in function1
at line8 in main.pl
```

I think both have their merits:
1. The topmost stack helps us understand what are the effects of the error (what will get impacted)
2. The deepest stack helps us understand the root cause

So whichever has more value can be used / preserved

Reply:
So, in your conclusion, which one is better ?

Reply:
I usually look for deepest first (the line which throws error) and then from where it  get called. stack trace normally last in first out

Reply:
yeah it is more helpful I feel

Reply:
Ok will fix it.

Reply:
Fix it @sasikala-binary @shantanu-deriv Please review again.